File: src/Talabi.Core/Entities/CustomerNotification.cs
namespace Talabi.Core.Entities;

public class CustomerNotification : BaseEntity
{
    public Guid CustomerId { get; set; }
    public Customer? Customer { get; set; }

    public string Title { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public string Type { get; set; } = "general"; // "OrderStatusChanged", "OrderDelivered", "Promotion", etc.

    public bool IsRead { get; set; }
    public DateTime? ReadAt { get; set; }

    public Guid? OrderId { get; set; }
    public Order? Order { get; set; }
}

File: mobile/lib/models/customer_notification.dart
class CustomerNotification {
  CustomerNotification({
    required this.id,
    required this.title,
    required this.message,
    required this.type,
    required this.isRead,
    required this.createdAt,
    this.readAt,
    this.relatedEntityId,
  });

  factory CustomerNotification.fromJson(Map<String, dynamic> json) {
    return CustomerNotification(
      id: json['id'].toString(),
      title: json['title'] ?? '',
      message: json['message'] ?? '',
      type: json['type'] ?? 'general',
      isRead: json['isRead'] ?? false,
      createdAt: DateTime.parse(json['createdAt']),
      readAt: json['readAt'] != null ? DateTime.parse(json['readAt']) : null,
      relatedEntityId: json['relatedEntityId']?.toString(),
    );
  }
  CustomerNotification copyWith({
    String? id,
    String? title,
    String? message,
    String? type,
    bool? isRead,
    DateTime? createdAt,
    DateTime? readAt,
    String? relatedEntityId,
  }) {
    return CustomerNotification(
      id: id ?? this.id,
      title: title ?? this.title,
      message: message ?? this.message,
      type: type ?? this.type,
      isRead: isRead ?? this.isRead,
      createdAt: createdAt ?? this.createdAt,
      readAt: readAt ?? this.readAt,
      relatedEntityId: relatedEntityId ?? this.relatedEntityId,
    );
  }

  final String id;
  final String title;
  final String message;
  final String type;
  final bool isRead;
  final DateTime createdAt;
  final DateTime? readAt;
  final String? relatedEntityId;
}
